test/system/src/test/java/io/pravega/test/system/SegmentReaderAPITest.java

```diff
@@ -0,0 +1,485 @@
+     * segment1 and segment2 has scaled down (segment3).
+     */
+    @Test(timeout = 90000)
+    public void getNextStreamCutWithScaleDownTest() throws SegmentTruncatedException, ExecutionException, InterruptedException {
+        String streamName = "testStreamSegmentScaleDown";
+        String streamScope = "testScopeSegmentScaleDown";
+        String readerGroupName = "testReaderGroupSegmentScaleDown";
+        String readerName = UUID.randomUUID().toString();
+        AtomicLong clock = new AtomicLong();
+        long approxDistanceToNextOffset = 180L;
+
+        StreamConfiguration streamConfiguration = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build();
+
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+        @Cleanup
+        ConnectionFactory connectionFactory = new SocketConnectionFactoryImpl(clientConfig);
+        ControllerImpl controller = new ControllerImpl(ControllerImplConfig.builder().clientConfig(clientConfig).build(),
+                connectionFactory.getInternalExecutor());
+        Stream stream = Stream.of(streamScope, streamName);
+
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(streamScope, controller, connectionFactory);
+        log.info("Invoking Writer test with Controller URI: {}", controllerURI);
+
+        StreamManager streamManager = StreamManager.create(clientConfig);
+        assertTrue("Creating Scope", streamManager.createScope(streamScope));
+        assertTrue("Creating stream", streamManager.createStream(streamScope, streamName, streamConfiguration));
+
+        @Cleanup
+        BatchClientFactory batchClient = BatchClientFactory.withScope(streamScope, clientConfig);
+
+        StreamCut streamCut0 = streamManager.fetchStreamInfo(streamScope, streamName).join().getHeadStreamCut();
+        log.info("Initial stream streamCut0 {}", streamCut0);
+        assertEquals(1, streamCut0.asImpl().getPositions().size());
+
+        List<Segment> list = streamCut0.asImpl().getPositions().keySet().stream().collect(Collectors.toList());
+
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName, new UTF8StringSerializer(),
+                EventWriterConfig.builder().build());
+
+        // write events to stream 30 * 5  = 150 bytes
+        writeEvents(5, writer);
+        writer.flush();
+
+        //Requested next stream cut at a distance of 180 bytes, and getting the next approx offset as a response.
+        StreamCut streamCut1 = batchClient.getNextStreamCut(streamCut0, approxDistanceToNextOffset);
+        long streamCut1Position = streamCut1.asImpl().getPositions().get(list.get(0)).longValue();
+        log.info("Next stream cut1 {} streamCut1 position {}", streamCut1, streamCut1Position);
+        assertTrue(150 <= streamCut1Position);
```

Comment:
Since we have written only 5events, can we make this assertion as equals to 150 as this would be the tail offset for the segment.

Reply:
made the change